pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -191,6 +201,12 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 		reqLogger.Info("Checking if cluster can commence upgrade.")
 		schedulerResult := r.scheduler.IsReadyToUpgrade(instance, cfg.GetUpgradeWindowTimeOutDuration())
 		if schedulerResult.IsReady {
+
+			err = eventClient.Notify(notifier.StateStarted)
```

Comment:
I feel like this should be part of the osd upgrader. Main reason is that It seems like a 'platform' specific action to me which the upgrader abstraction is for. 
This way if a user of MUO wants to avoid sending notifications they can create a new upgrader which omits notifications.

Reply:
I've moved this into the OSD upgrader and out of the controller now. Not written any tests for the new steps yet. Do you want to take a quick look before I squash the commits and see what you think?